app/controllers/guests_controller.rb

```diff
@@ -0,0 +1,12 @@
+class GuestsController < ApplicationController
+    def show
+        @guest = Guest.find(params[:id])
+    end
+
+    def update
+        @guest = Guest.find(params[:id])
+        @guest.add_room(params[:room_id])
+        
+        redirect_to "/guests/#{params[:id]}"
+    end
```

Comment:
After looking through your codebase, it looks like you're using this to associate a guest with a room. And you're doing this in an `update` in the controller. If we imagine what we're doing on the database, we are CREATING a relationship between these two resources and adding a new row to the `guest_rooms` join table. Those are the two most important pieces to consider, we are creating a new row, which is going to mean this is going to be a POST, and we are creating a new row on the joins table, meaning that this request should go to the associated joins controller.

```ruby
class GuestRoomsController < ApplicationController
  def create
    guest = Guest.find(params[:id])
    room = Room.find(params[:room_id])

    GuestRoom.create(guest: guest, room: room)

    redirect_to "/guests/#{params[:id]}"
  end
end
```

Only other thing is that you do not need to use instance variables if you are not sending those variables to a view.